package/MDAnalysis/analysis/bat.py

```diff
@@ -197,7 +197,11 @@ def _sort_atoms_by_mass(atoms, reverse=False):
     ag_n : list of Atoms
         Sorted list
     """
-    return sorted(atoms, key=lambda a: (a.mass, a.index), reverse=reverse)
+    s = sorted(atoms, key=lambda a: (a.mass, a.index))
+    if reverse:
+        s.reverse()
+
+    return s
 
```

Comment:
I added this part to the bat.py cause the `reverse` parameter of the `sorted` method doesn't work as expected with `nan` values of masses, which causes errors in sorting

Reply:
I discovered this issue when the universe in the `test_bat::test_bat_root_selection` test gave weird output.  I found atoms of this universe have all masses equal to `nan` (which used to be zero before failing the test). After investigation, the problem seemed to be that the reverse parameter of the `sorted` function doesn't give any effect. So, for testing this I'd use the same universe which will end up writing the same test_bat_root_selection, or am I missing something?

Reply:
Please add a comment to the source code to explain the behavior — this exactly the kind of hard-won insight that should be documented. Otherwise, the next person will "optimize" the code back to the previous version.

Reply:
I honestly don't remember what error influenced me to make this change. I re-tested the old behavior and it works fine indeed, so I'll remove this change